FreeRTOS/Test/CMock/stream_buffer/stream_buffer_utest.c

```diff
@@ -455,40 +488,44 @@ void test_xStreamBufferCreate_malloc_fail( void )
 }
 
 /**
- * @brief Assertion fails if a zero buffer size is passed as  the parameter.
+ * @brief Assertion fails if a zero buffer size is passed as the xBufferSizeBytes parameter.
  */
 void test_xStreamBufferCreate_zero_buffer_size( void )
 {
+    xStreamBuffer = xStreamBufferCreate( 0, TEST_STREAM_BUFFER_TRIGGER_LEVEL );
 
-    if( TEST_PROTECT() )
-    {
-        ( void ) xStreamBufferCreate( 0, TEST_STREAM_BUFFER_TRIGGER_LEVEL );
-    }
-    validate_and_clear_assertions();
+    validate_and_clear_multiple_assertions();
```

Comment:
This test case specifically checks if assertion is called and program is aborted on passing a zero size. I think we shouldn't continue the program execution and check for other assertions. Here, the test case is trying to do multiple things.

Reply:
Asserts can be turned off in production. This tests both the configASSERT disabled and configASSERT enabled cases.